### ISSUE TITLE
Make currently used `sed` commands compatible with FreeBSD systems

### ISSUE THREAD

**Issue by rdiachenko** [checkstyle/checkstyle#13078]
Detected in https://github.com/checkstyle/checkstyle/pull/13064#discussion_r1193996535

`-n`
We need to review existing scripts where `sed` is used and add flag `-n` where it is needed to make it behave on Mac the same way as it currently behaves on non FreeBSD systems.

`-e`
No need for this flag in cases with a single command.

`-E`
Use extended regular expressions for complex sed commands because some of the expressions don't work without `-E` on FreeBSD systems (e.g., #13063)

**Comment by achyuth8055** [comment]
Hello @romani ,

I’m a new contributor exploring beginner-friendly issues and would like to work on this issue if it’s still available.
Please let me know if there are any specific guidelines or files I should focus on before starting.

Thank you!

**Comment by romani** [comment]
yes, still bunch of `sed` in scripts and they likely not working well in macos

**Comment by achyuth8055** [comment]
Thanks for confirming @romani 

I can start working on this Could you point me to the main scripts or directories where `sed` usage is most problematic on macOS? Also, should the goal be to make all `sed` commands compatible with BSD/macOS `sed` or is it acceptable to conditionally handle GNU and BSD differences?

That will help me approach the fixes correctly.
